contracts/lib/OrderCombiner.sol

```diff
@@ -175,6 +180,32 @@ contract OrderCombiner is OrderFulfiller, FulfillmentApplier {
             mstore(orderHashes, 0)
         }
 
+        // Declare an error buffer indicating status of any native offer items.
+        // {00} == 0 => In a match function, no native offer items: allow.
+        // {01} == 1 => In a match function, some native offer items: allow.
+        // {10} == 2 => Not in a match function, no native offer items: allow.
+        // {11} == 3 => Not in a match function, some native offer items: THROW.
+        uint256 invalidNativeOfferItemErrorBuffer;
+
+        // Use assembly to set the value for the second bit of the error buffer.
+        assembly {
+            // Use the second bit of the error buffer to indicate whether the
+            // current function is not matchAdvancedOrders or matchOrders.
+            invalidNativeOfferItemErrorBuffer := shl(
+                1,
+                gt(
+                    // Take the remainder of the selector modulo a magic value.
```

Comment:
:eyes:

Which (two) selectors is this comparing? This seems like an extremely fragile piece of code should new external code paths allowed to reach this function.

Reply:
It seems to be reachable from `Consideration.matchOrders`, ` Consideration.matchAdvancedOrders`, `Consideration.fulfillAvailableOrders`, `Consideration.fulfillAvailableAdvancedOrders`.

That is 4 selectors this code is comparing. Note that not only new external functions are risky here, but also any parameter change to the above would change the selectors.

Reply:
Found a prior conversation on the topic: https://github.com/ProjectOpenSea/seaport/pull/464/files#r893417290

Reply:
My recommendation would be to pass down a `isMatchOrder` flag.

Reply:
Yeah we’ll definitely replace this pattern with one like you suggest should any aspect of the interface change